processing/src/test/java/io/druid/query/lookup/LookupSnapshotTakerTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.lookup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import io.druid.segment.TestHelper;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.util.List;
+
+
+public class LookupSnapshotTakerTest
+{
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private final ObjectMapper mapper = TestHelper.JSON_MAPPER;
+
+
+  private LookupSnapshotTaker lookupSnapshotTaker;
+  private String basePersistDirecotry;
+
+  @Before
+  public void setUp() throws IOException
+  {
+    basePersistDirecotry = temporaryFolder.newFolder().getAbsolutePath();
+    lookupSnapshotTaker = new LookupSnapshotTaker(mapper, basePersistDirecotry);
+    mapper.registerSubtypes(LookupFactoryMock.class);
+    mapper.registerSubtypes(LookupExtractorFactory.class);
+    mapper.addMixIn(LookupExtractorFactory.class, LookupFactoryMock.class);
+  }
+
+  @After
+  public void tearDown()
+  {
+    temporaryFolder.delete();
+  }
+
+  @Test
+  public void testTakeSnapshotAndPullExisting() throws IOException
+  {
+
+    LookupBean lookupBean = new LookupBean();
+    lookupBean.name ="name";
+    lookupBean.factory = new LookupFactoryMock();
+    List<LookupBean> lookupBeanList = Lists.newArrayList(lookupBean);
+    lookupSnapshotTaker.takeSnapshot(lookupBeanList);
+    List<LookupBean> actualList = lookupSnapshotTaker.pullExistingSnapshot();
+    Assert.assertEquals(lookupBeanList, actualList);
+  }
+
+  @JsonTypeName("mock")
+  private class LookupFactoryMock implements LookupExtractorFactory
```

Comment:
Actually, in the other PR for QTL config I added a MapLookupExtractorFactory which might fit better than a mock.